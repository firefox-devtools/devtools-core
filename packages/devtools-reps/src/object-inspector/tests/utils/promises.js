/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

const {
  getPromiseProperties,
  isPromise,
} = require("../../utils");

describe("promises utils function", () => {
  it("is promise", () => {
    const promise = {
      contents: {
        enumerable: true,
        configurable: false,
        value: {
          frozen: false,
          ownPropertyLength: 0,
          preview: {
            kind: "Object",
            ownProperties: {},
            ownPropertiesLength: 0,
            safeGetterValues: {}
          },
          actor: "server2.conn2.child1/pausedobj36",
          promiseState: {
            state: "rejected",
            reason: {
              type: "undefined"
            },
            creationTimestamp: 1486584316133.3994,
            timeToSettle: 0.001713000237941742
          },
          class: "Promise",
          type: "object",
          extensible: true,
          sealed: false
        },
        writable: true
      }
    };

    expect(isPromise(promise)).toEqual(true);
  });

  it("getPromiseProperties", () => {
    const promise = {
      path: "root",
      contents: {
        enumerable: true,
        configurable: false,
        value: {
          frozen: false,
          ownPropertyLength: 0,
          preview: {
            kind: "Object",
            ownProperties: {},
            ownPropertiesLength: 0,
            safeGetterValues: {}
          },
          actor: "server2.conn2.child1/pausedobj36",
          promiseState: {
            state: "rejected",
            reason: {
              type: "3"
            },
            creationTimestamp: 1486584316133.3994,
            timeToSettle: 0.001713000237941742
          },
          class: "Promise",
          type: "object",
          extensible: true,
          sealed: false
        },
        writable: true
      }
    };

    const properties = getPromiseProperties(promise);
    expect(properties).toMatchSnapshot();
  });
});
