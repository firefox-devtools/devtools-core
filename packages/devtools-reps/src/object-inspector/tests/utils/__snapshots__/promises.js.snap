// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`promises utils function makeNodesForPromiseProperties 1`] = `
Array [
  Object {
    "contents": Object {
      "value": "rejected",
    },
    "name": "<state>",
    "parent": Object {
      "contents": Object {
        "value": Object {
          "actor": "server2.conn2.child1/pausedobj36",
          "class": "Promise",
          "promiseState": Object {
            "reason": Object {
              "type": "3",
            },
            "state": "rejected",
          },
          "type": "object",
        },
      },
      "path": "root",
    },
    "path": "root/##-state",
    "type": Symbol(<state>),
  },
  Object {
    "contents": Object {
      "value": Object {
        "type": "3",
      },
    },
    "name": "<reason>",
    "parent": Object {
      "contents": Object {
        "value": Object {
          "actor": "server2.conn2.child1/pausedobj36",
          "class": "Promise",
          "promiseState": Object {
            "reason": Object {
              "type": "3",
            },
            "state": "rejected",
          },
          "type": "object",
        },
      },
      "path": "root",
    },
    "path": "root/##-reason",
    "type": Symbol(<reason>),
  },
]
`;
